app/src/main/java/org/mozilla/fenix/perf/Performance.kt

```diff
@@ -37,4 +45,25 @@ object Performance {
             activity.reportFullyDrawn()
         }
     }
+
+    /**
+     * Processes intent for Performance testing to remove protection pop up ( but keeps the TP
+     * on) and removes the onboarding screen.
+     */
+    fun processIntentIfPerformanceTest(intent: Intent, context: Context) {
+        if (!intent.getBooleanExtra(EXTRA_IS_PERFORMANCE_TEST, true)) {
+            return
+        }
+
+        val batteryStatus = context.registerReceiver(null, IntentFilter(Intent.ACTION_BATTERY_CHANGED))
+        val isPhonePlugged = batteryStatus!!.getIntExtra(BatteryManager.EXTRA_PLUGGED, -1) ==
```

Comment:
Can we handle the case if this is null? Other apps could add this extra to trigger this intent so we should make it handled gracefully.